staging/src/k8s.io/apiserver/pkg/authentication/request/headerrequest/requestheader.go

```diff
@@ -126,10 +126,12 @@ func (a *requestHeaderAuthRequestHandler) AuthenticateRequest(req *http.Request)
 		}
 	}
 
-	return &user.DefaultInfo{
-		Name:   name,
-		Groups: groups,
-		Extra:  extra,
+	return &authenticator.Response{
```

Comment:
TODO to take implicit audiences at requestHeaderAuthRequestHandler construction time, and if ctx is audience-limited, ensure it intersects with the implicit audiences, and echo an audience in the response